server/src/main/java/io/crate/planner/node/ddl/AlterTableAddColumnPlan.java

```diff
@@ -130,6 +108,51 @@ public void executeOrFail(DependencyCarrier dependencies,
         }
     }
 
+    public static AddColumnRequest createRequest(AnalyzedAlterTableAddColumn alterTable,
+                                                 NodeContext nodeContext,
+                                                 PlannerContext plannerContext,
+                                                 Row params,
+                                                 SubQueryResults subQueryResults,
+                                                 FulltextAnalyzerResolver fulltextAnalyzerResolver) {
+        AnalyzedTableElements<Object> tableElements = validate(
+            alterTable,
+            plannerContext.transactionContext(),
+            nodeContext,
+            params,
+            subQueryResults
+        );
+
+        AnalyzedTableElements.validateAndBuildSettings(tableElements, fulltextAnalyzerResolver);
+
+        AnalyzedTableElements.finalizeAndValidate(
+            alterTable.tableInfo().ident(),
+            alterTable.analyzedTableElementsWithExpressions(),
+            tableElements
+        );
+
+        // We can add multiple object columns via ALTER TABLE ADD COLUMN.
+        // Those columns can have overlapping paths, for example we can add columns o['a']['b'] and o['a']['c'].
+        // For every added column AnalyzedColumnDefinition provides not only leaf but also path to the root.
+        // For o['a']['b'] and o['a']['c'] we can end up having Ref(Ident(o)) and Ref(Ident(a)) twice.
+
+        // We need a Map to compare References by FQN.
+        // Regular Set cannot be used as it would use Reference.position along with other fields when calling equals() and
+        // position is resolved to -1 and -2 for all parts of the path in the case above so overlapping parts will be "different".
+
+        // pKeyIndices is constructed based on insertion order, so we need a LinkedHashMap.
+
+        LinkedHashMap<ColumnIdent, Reference> references = new LinkedHashMap<>();
+        IntArrayList pKeysIndices = new IntArrayList();
+        tableElements.collectReferences(alterTable.tableInfo().ident(), references, pKeysIndices, true);
+
+        return new AddColumnRequest(
+            alterTable.tableInfo().ident(),
+            List.copyOf(references.values()), // We don't use Map in the request itself since we need directly indexed structure referred by pKeysIndices.
```

Comment:
If we keep the unmodifiable approach, why not `Collections.unmodifiableList()` ?

Reply:
We don't expect changes in references after creating a request, so no need to create a view?

Reply:
Personally, especially for non-critical code path (running for each row) it's good to pass around immutable structures, when you don't want some part of the code later on to accidentally make changes, but let's hear @mfussenegger opinion on that.

Reply:
I specifically removed some of the copy operations here https://github.com/crate/crate/commit/3f71aac83dcba3bcedc3d4c61ec64efae6be6081

Reply:
So let's revert https://github.com/crate/crate/commit/34c3244c0cb2a4a3f956678546f570bb9a34c7e5 and we don't need to protect exposed refs via extra copy? I mean `references()` which is used outside of the request.

Reply:
fine for me as well

Reply:
I'd drop it